sysown/proxysql
# unit8_t conversion

Description:
converted charset/default_charset data type to unsigned int. according to `typedef struct ma_charset_info_st`

Testing:
1. SET NAMES
- Use `set names utf8mb4 collate utf8mb4_croatian_ci` through proxy.
- Check `show variables like 'collation_connection'`. Collation is set to `utf8mb4_croatian_ci` with 

2. Setting charset in admin
- Use admin and set `update global_variables set variable_value='latin1' where variable_name='mysql-default_charset'` and update runtime;
- Use admin and set `update global_variables set variable_value='utf8mb4' where variable_name='mysql-default_charset'` and update runtime.
- Login to proxysql and check collation `show variables like 'collation_connection'`. Collation is set to `utf8mb4_general_ci` because proxysql changed it. (for mysql version < 8.0)
- Login to proxysql and check collation `show variables like 'collation_connection'`. Collation is set to `utf8mb4_0900_ai_ci`.  (for mysql version >= 8.0)

3. Start client with `--default_character_set=utf8mb4`
- Connecting to mysqld version < 8.0
```
mysql> show variables like 'collation_connection';
+----------------------+--------------------+
| Variable_name        | Value              |
+----------------------+--------------------+
| collation_connection | utf8mb4_general_ci |
+----------------------+--------------------+
3 rows in set (0.00 sec)
```

- Connecting to mysqld version >= 8.0
```
mysql> show variables like 'collation_connection';
+-------------------------------+--------------------+
| Variable_name                 | Value              |
+-------------------------------+--------------------+
| collation_connection          | utf8mb4_0900_ai_ci |
+-------------------------------+--------------------+
4 rows in set (0.00 sec)
```
